src/App.js

```diff
@@ -17,15 +17,20 @@ const useSemiPersistentState = () => {
 function App() {
   const [todoList, setTodoList] = useSemiPersistentState();
   
-  function addTodo(newTodo) {
-    setTodoList([...todoList, newTodo])
-  }
+  const addTodo = (newTodo) => {
+    setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
+ };
+
+  function removeTodo(id) {
+    let filteredArray = (todoList.filter(todo => todo.id !== id))
+    setTodoList(filteredArray)
+  };
```

Comment:
a little filter don't hurt nobody :)

You can actually use a `const` here since `filteredArray` isn't being reused again.

Also, remember what we did for setting the todolist in the `addTodo` function. We'll need the previous state to create the new state so we should do the filter with the `setTodoList` again. I'll let you figure out how to do this 😄